### PR TITLE
Fix problems in on_pfs_update

### PR DESCRIPTION
**Pull request by palango**
It's really strange that this stuff wasn't found by mypy. I guess we should look into #167 soon.

### CODE REVIEW

File: src/raiden_libs/matrix.py [hunk 1]
```diff
@@ -178,7 +179,7 @@ def _handle_message(self, room, event) -> bool:
             try:
                 message_dict = json.loads(line)
                 message = message_from_dict(message_dict)
-            except (UnicodeDecodeError, json.JSONDecodeError) as ex:
+            except (UnicodeDecodeError, json.JSONDecodeError, KeyError) as ex:
                 logger.warning(
                     "Can't parse message data JSON",
```
Comment (karlb) [inline src/raiden_libs/matrix.py]: The error message is a bit misleading in case of a `KeyError`. Still somewhat ok, though.

Comment (palango) [reply src/raiden_libs/matrix.py]: True, I'll add another case for this.

**Comment by palango** [comment]
I still don't get why mypy doesn't complain about this:

- Our `Address` type is defined as `Address = NewType('Address', str)`
- This is used in the tests like this, where `addresses` has the type `List[Address]`:
```
    message = get_updatepfs_message(
        updating_participant=addresses[0],
        other_participant=addresses[1],
        updating_capacity=TokenAmount(UINT256_MAX + 1),
    )
```
- In there a `UpdatePFS` message is created, where `Address = NewType('Address', bytes)`
```
class UpdatePFS(SignedMessage):
    def __init__(
            self,
            *,
            canonical_identifier: CanonicalIdentifier,
            updating_participant: typing.Address,
            other_participant: typing.Address,
```

For me mypy should be able to see through this.

**Comment by karlb** [comment]
Maybe two `NewType`s with the same name are treated as the same type?

**Comment by karlb** [comment]
> Maybe two `NewType`s with the same name are treated as the same type?

No, that's not generally the case. The following program correctly complains:

```python
from typing import NewType


class C1:
    Address = NewType('Address', str)


class C2:
    Address = NewType('Address', bytes)


def f(x: C1.Address):
    return x


byte_addr = C2.Address(bytes([1, 2]))
f(byte_addr)
```

**Comment by palango** [comment]
Seems like the problem is the `--ignore-missing-imports` flag. When I take it out of the mypy invocation I get (amoing other) the following warning:
```
src/monitoring_service/service.py:19: error: Cannot find module named 'raiden.utils.typing'
```

**Comment by karlb** [comment]
> No the question is why that is.

I think mypy ignores that package because it doesn't declare itself as properly typed via PEP561:
https://mypy.readthedocs.io/en/latest/installed_packages.html#using-pep-561-compatible-packages-with-mypy

**Comment by karlb** [comment]
See https://github.com/raiden-network/raiden/pull/3698. I'll try to fix as many of the type failures caused by this as possible before that PR is merged.

File: src/raiden_libs/matrix.py (hunk 1 above)
Comment (palango) [reply src/raiden_libs/matrix.py]: Moved it to the next except block which fits better.

**Pull request merged**